RELEASE_PROCESS.md

```diff
@@ -0,0 +1,200 @@
+
+## Releasing
+
+For the duration of this document, we will pretend the "golden commit" version is `8a8ee28`, the version is `0.3.1`, and our release date is 2016-01-15. Any place you see those values, please substitute your own version.
+
+### Update the GitHub issue
+
+Leave a comment on the GitHub issue for this release to let everyone know that you are starting.
+
+### Update VERSION and CHANGELOG
+
+* git pull
+* git checkout -b release-0.3.1
+
+#### Update the VERSION
+
+In the root of the ponyc repo is a file called `VERSION`. You should update the version number therein to your version number, for example: `0.3.1`.
+
+#### Update the CHANGELOG
+
+We maintain all changes in our CHANGELOG. At the time we do a release, we need to "roll" the CHANGELOG from one version to another. Open up CHANGELOG.md and you will see a series of changes that are in this commit under a section that says:
+
+`## [unreleased] - unreleased`
+
+This should be changed to:
+
+`## [0.3.1] - 2016-01-15`
+
+If any of the sections, "Added", "Fixed" or "Changed" is empty, delete it.
+
+### Merge into release
+
+* git checkout release
+* git merge release-0.3.1
+
+### Tag the release
+
+* git tag 0.3.1
+
+### Push to the release branch
+
+* git push origin
+* git push origin 0.3.1
+
+### Update CHANGELOG for new entries
+
+We are in a slightly precarious state here, someone might have merged since we started and updated the CHANGELOG on master. Verify that hasn't happened then merge your change to CHANGELOG into master.
+
+* git checkout master
+* git merge release-0.3.1
+
+Now add a new "unreleased" section to the change log. Above the `## [0.3.1] - 2016-01-15` add the following:
+
+```
+## [unreleased] - unreleased
+
+### Fixed
+
+### Added
+
+### Changed
+
+```
+
+If anything was merged into the CHANGELOG since versioned it, move the new additions to the appropriate section of the new "unreleased" section. Commit your changes and push to master.
+
+### Update the GitHub issue
+
+Leave a comment on the GitHub issue for this release to let everyone know you are done versioning the CHANGELOG and VERSION and that they are updated on `master`.
+
+### "Update" PRs for CHANGELOG
+
+Go through all open PRs and check to see if they are updating the CHANGELOG. If they are, leave a note asking them to merge the latest CHANGELOG into their branch because of the recent versioning for release and to move their additions into the appropriate area of "unreleased".
+
+### Add CHANGELOG entries to GitHub releases
+
+By now GitHub should have a listing of this new release under [releases](https://github.com/ponylang/ponyc/releases). Click the `0.3.1` then do "Edit Release". Paste the CHANGELOG entries for this release into the box and update.
+
+### Update Homebrew
+
+Fork the [homebrew-core repo](https://github.com/Homebrew/homebrew-core) and then clone it locally. You are going to be editing "Formula/ponyc.rb". Make sure you do your changes on a branch:
```

Comment:
People will do this more than once. Include notes about updating from remote master if the fork already exists